### PR TITLE
Support managing domains in the K8S application

### PR DESCRIPTION
**Pull request by Mahmoud-Emad**
### Description

The user might have some solutions on the cluster that require an ingress with a domain.

### Changes

- Optimized the `ManageGatewy` component to work on both types of machines normal `VM` and `K8S` machine
- Generate the name based on the master
- takes the master networks as the domain-supported networks
- Change the dialog title to be dynamic
- Working on refactoring the component
- Load the WireGuard network name dynamic based on the machine type

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3149
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3264

### Screenshots/Video

 [Screencast from 08-18-2024 03:10:58 PM.webm](https://github.com/user-attachments/assets/89de2baf-c59c-4849-9241-1e612823da9a)

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by MohamedElmdary — APPROVED** [review]

File: packages/playground/src/weblets/tf_deployment_list.vue [hunk 1]
```diff
@@ -464,6 +473,7 @@ async function onDelete(k8s = false) {
 
 const VMS: string[] = [ProjectName.Fullvm, ProjectName.VM, ProjectName.NodePilot];
 function openDialog(project: string, item?: any): void {
+  console.log({ item });
```
Comment (amiraabouhadid) [inline packages/playground/src/weblets/tf_deployment_list.vue]: why do we need this console.log?

File: yarn.lock [hunk 2]
```diff
@@ -7300,6 +7300,11 @@ follow-redirects@^1.0.0, follow-redirects@^1.14.7, follow-redirects@^1.14.9, fol
   resolved "https://registry.npmjs.org/follow-redirects/-/follow-redirects-1.15.2.tgz"
   integrity sha512-VQLG33o04KaQ8uYi2tVNbdrWp1QWxNNea+nmIB4EVM28v0hmP17z7aG1+wAkNzVq4KeXTq3221ye5qTJP91JwA==
 
+follow-redirects@^1.15.6:
```
Comment (amiraabouhadid) [inline yarn.lock]: why are there changes in this file?

Comment (ghost) [reply yarn.lock]: I just run yarn

**Review by samaradel — APPROVED** [review]

File: yarn.lock [hunk 3]
```diff
@@ -4563,12 +4563,12 @@ axios@0.25.0:
   dependencies:
     follow-redirects "^1.14.7"
 
-axios@1.5.0:
-  version "1.5.0"
-  resolved "https://registry.yarnpkg.com/axios/-/axios-1.5.0.tgz#f02e4af823e2e46a9768cfc74691fdd0517ea267"
-  integrity sha512-D4DdjDo5CY50Qms0qGQTTw6Q44jl7zRwY7bthds06pUGfChBCTcQs+N743eFWGEd6pRTMd6A+I87aWyFV5wiZQ==
+axios@1.7.4:
+  version "1.7.4"
```
Comment (amiraabouhadid) [inline yarn.lock]: please remove changes in this file as no new packages are required for the code above.

**Review by amiraabouhadid — CHANGES_REQUESTED** [review]

**Review by amiraabouhadid — APPROVED** [review]

**Pull request merged**